Tasks/Common/webdeployment-common/msdeployutility.ts

```diff
@@ -98,45 +99,67 @@ export function getMSDeployCmdArgs(webAppPackage: string, webAppName: string, pu
  * @returns boolean
  */
 export async  function containsParamFile(webAppPackage: string ) {
-    var parameterFile = tl.getVariable('System.DefaultWorkingDirectory') + '\\' + 'parameter.xml';
-    var fd = fs.openSync(parameterFile, "w");
-    var outputObj = fs.createWriteStream("",{"fd": fd});
-    
+    var isParamFilePresent = false;
+    var isParamFilePresentBackup;
     try {
-        var msDeployCheckParamFileCmdArgs = "-verb:getParameters -source:package=\'" + webAppPackage + "\'";
-        await tl.exec("msdeploy", msDeployCheckParamFileCmdArgs, <any>{ failOnStdErr: true, outStream: outputObj });
+        isParamFilePresentBackup = await isMSDeployPackage(webAppPackage);
     }
-    catch(error) {
-        throw Error(error);
+    catch (error) {
+        isParamFilePresentBackup = false;
+        tl.debug("an error occurred in the function isMSDeployPackage");
     }
-    finally {
-        fs.fsyncSync(fd);
-        fs.closeSync(fd);
+    var msDeployCheckParamFileCmdArgs = "-verb:getParameters -source:package=\'" + webAppPackage + "\'";
+    var res =  tl.execSync("msdeploy", msDeployCheckParamFileCmdArgs, <any>{ failOnStdErr: true });
+
+    if((res.error && res.error.message.length > 0 ) || (res.stderr && res.stderr.length > 0)) {
+        tl.debug("[COMMAND FAILED]: msdeploy " + msDeployCheckParamFileCmdArgs);
+        if(res.error) {
+            tl.debug("res.error: " + res.error.message);
```

Comment:
Message will not be null because it will always have the output of the error stream. And if the msdeploy command has failed, then we can be sure that there is something in the error stream